File: Source/Contexts/Theme.context.ts
import Roact from "@rbxts/roact";

import { pColor } from "@rbxts/precomputed";

/*
: {
    50: Color3.fromHex(""),
    100: Color3.fromHex(""),
    200: Color3.fromHex(""),
    300: Color3.fromHex(""),
    400: Color3.fromHex(""),
    500: Color3.fromHex(""),
    600: Color3.fromHex(""),
    700: Color3.fromHex(""),
    800: Color3.fromHex(""),
    900: Color3.fromHex("")
},
*/

export const ThemeContext = Roact.createContext({
	Layout: {
        Background: pColor.Black,
        Foreground: identity<Color3 | undefined>(undefined),
        Seperator: identity<Color3 | undefined>(undefined),
        Focus: identity<Color3 | undefined>(undefined)
    },
    Content: {
        ContentA: identity<Color3 | undefined>(undefined),
        ContentB: identity<Color3 | undefined>(undefined),
        ContentC: identity<Color3 | undefined>(undefined),
        ContentD: identity<Color3 | undefined>(undefined)
    },
    Base: {
        Default: identity<Color3 | undefined>(undefined),
        Primary: identity<Color3 | undefined>(undefined),
        Secondary: identity<Color3 | undefined>(undefined),
        Success: identity<Color3 | undefined>(undefined),
        Warning: identity<Color3 | undefined>(undefined),
        Danger: identity<Color3 | undefined>(undefined)
    },
    Default: {
        Default50: identity<Color3 | undefined>(undefined),
        Default100: identity<Color3 | undefined>(undefined),
        Default200: identity<Color3 | undefined>(undefined),
        Default300: identity<Color3 | undefined>(undefined),
        Default400: identity<Color3 | undefined>(undefined),
        Default500: identity<Color3 | undefined>(undefined),
        Default600: identity<Color3 | undefined>(undefined),
        Default700: identity<Color3 | undefined>(undefined),
        Default800: identity<Color3 | undefined>(undefined),
        Default900: identity<Color3 | undefined>(undefined)
    },
    
    Primary: {
        Primary50: identity<Color3 | undefined>(undefined),
        Primary100: identity<Color3 | undefined>(undefined),
        Primary200: identity<Color3 | undefined>(undefined),
        Primary300: identity<Color3 | undefined>(undefined),
        Primary400: identity<Color3 | undefined>(undefined),
        Primary500: identity<Color3 | undefined>(undefined),
        Primary600: identity<Color3 | undefined>(undefined),
        Primary700: identity<Color3 | undefined>(undefined),
        Primary800: identity<Color3 | undefined>(undefined),
        Primary900: identity<Color3 | undefined>(undefined)
    },
    Secondary: {
        Secondary50: identity<Color3 | undefined>(undefined),
        Secondary100: identity<Color3 | undefined>(undefined),
        Secondary200: identity<Color3 | undefined>(undefined),
        Secondary300: identity<Color3 | undefined>(undefined),
        Secondary400: identity<Color3 | undefined>(undefined),
        Secondary500: identity<Color3 | undefined>(undefined),
        Secondary600: identity<Color3 | undefined>(undefined),
        Secondary700: identity<Color3 | undefined>(undefined),
        Secondary800: identity<Color3 | undefined>(undefined),
        Secondary900: identity<Color3 | undefined>(undefined)
    },
    Success: {
        Success50: identity<Color3 | undefined>(undefined),
        Success100: identity<Color3 | undefined>(undefined),
        Success200: identity<Color3 | undefined>(undefined),
        Success300: identity<Color3 | undefined>(undefined),
        Success400: identity<Color3 | undefined>(undefined),
        Success500: identity<Color3 | undefined>(undefined),
        Success600: identity<Color3 | undefined>(undefined),
        Success700: identity<Color3 | undefined>(undefined),
        Success800: identity<Color3 | undefined>(undefined),
        Success900: identity<Color3 | undefined>(undefined)
    },
    Warning: {
        Warning50: identity<Color3 | undefined>(undefined),
        Warning100: identity<Color3 | undefined>(undefined),
        Warning200: identity<Color3 | undefined>(undefined),
        Warning300: identity<Color3 | undefined>(undefined),
        Warning400: identity<Color3 | undefined>(undefined),
        Warning500: identity<Color3 | undefined>(undefined),
        Warning600: identity<Color3 | undefined>(undefined),
        Warning700: identity<Color3 | undefined>(undefined),
        Warning800: identity<Color3 | undefined>(undefined),
        Warning900: identity<Color3 | undefined>(undefined)
    },
    Danger: {
        Danger50: identity<Color3 | undefined>(undefined),
        Danger100: identity<Color3 | undefined>(undefined),
        Danger200: identity<Color3 | undefined>(undefined),
        Danger300: identity<Color3 | undefined>(undefined),
        Danger400: identity<Color3 | undefined>(undefined),
        Danger500: identity<Color3 | undefined>(undefined),
        Danger600: identity<Color3 | undefined>(undefined),
        Danger700: identity<Color3 | undefined>(undefined),
        Danger800: identity<Color3 | undefined>(undefined),
        Danger900: identity<Color3 | undefined>(undefined)
    }
});

File: Output/Contexts/Theme.context.lua
-- Compiled with roblox-ts v2.2.0
local TS = _G[script]
local Roact = TS.import(script, TS.getModule(script, "@rbxts", "roact").src)
local pColor = TS.import(script, TS.getModule(script, "@rbxts", "precomputed").out).pColor
--[[
	
	: {
	    50: Color3.fromHex(""),
	    100: Color3.fromHex(""),
	    200: Color3.fromHex(""),
	    300: Color3.fromHex(""),
	    400: Color3.fromHex(""),
	    500: Color3.fromHex(""),
	    600: Color3.fromHex(""),
	    700: Color3.fromHex(""),
	    800: Color3.fromHex(""),
	    900: Color3.fromHex("")
	},
	
]]
local ThemeContext = Roact.createContext({
	Layout = {
		Background = pColor.Black,
		Foreground = nil,
		Seperator = nil,
		Focus = nil,
	},
	Content = {
		ContentA = nil,
		ContentB = nil,
		ContentC = nil,
		ContentD = nil,
	},
	Base = {
		Default = nil,
		Primary = nil,
		Secondary = nil,
		Success = nil,
		Warning = nil,
		Danger = nil,
	},
	Default = {
		Default50 = nil,
		Default100 = nil,
		Default200 = nil,
		Default300 = nil,
		Default400 = nil,
		Default500 = nil,
		Default600 = nil,
		Default700 = nil,
		Default800 = nil,
		Default900 = nil,
	},
	Primary = {
		Primary50 = nil,
		Primary100 = nil,
		Primary200 = nil,
		Primary300 = nil,
		Primary400 = nil,
		Primary500 = nil,
		Primary600 = nil,
		Primary700 = nil,
		Primary800 = nil,
		Primary900 = nil,
	},
	Secondary = {
		Secondary50 = nil,
		Secondary100 = nil,
		Secondary200 = nil,
		Secondary300 = nil,
		Secondary400 = nil,
		Secondary500 = nil,
		Secondary600 = nil,
		Secondary700 = nil,
		Secondary800 = nil,
		Secondary900 = nil,
	},
	Success = {
		Success50 = nil,
		Success100 = nil,
		Success200 = nil,
		Success300 = nil,
		Success400 = nil,
		Success500 = nil,
		Success600 = nil,
		Success700 = nil,
		Success800 = nil,
		Success900 = nil,
	},
	Warning = {
		Warning50 = nil,
		Warning100 = nil,
		Warning200 = nil,
		Warning300 = nil,
		Warning400 = nil,
		Warning500 = nil,
		Warning600 = nil,
		Warning700 = nil,
		Warning800 = nil,
		Warning900 = nil,
	},
	Danger = {
		Danger50 = nil,
		Danger100 = nil,
		Danger200 = nil,
		Danger300 = nil,
		Danger400 = nil,
		Danger500 = nil,
		Danger600 = nil,
		Danger700 = nil,
		Danger800 = nil,
		Danger900 = nil,
	},
})
return {
	ThemeContext = ThemeContext,
}
